### PR TITLE
Fix Test_IngestMultipleTypesIntoOneColumn  test

### PR DESCRIPTION
**Pull request by sunitakawane**
# Description,

Test_IngestMultipleTypesIntoOneColumn .It was using ProcessIndexRequest with nil return value and working before. When I changed the ProcessIndexRequest implementation to use ple , it is failing doLogEventFilling: unknown ctype: 1"
It was not handling bool type

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/writer/segwriter.go [hunk 1]
```diff
@@ -356,6 +356,23 @@ func (ss *SegStore) doLogEventFilling(ple *ParsedLogEvent, tsKey *string) (bool,
...
+			startIdx := colWip.cbufidx
+			copy(colWip.cbuf[startIdx:], ple.allCvalsTypeLen[i][0:1])
+			colWip.cbufidx += 1
+			copy(colWip.cbuf[colWip.cbufidx:], ple.allCvals[i][0:1])
+			colWip.cbufidx += 1
+			boolVal := utils.BytesToBoolLittleEndian(ple.allCvals[i][0:1])
+			var asciiBytesBuf bytes.Buffer
+			_, err := fmt.Fprintf(&asciiBytesBuf, "%v", boolVal)
+			if err != nil {
+				return false, utils.TeeErrorf("doLogEventFilling: cannot write boolVal %v: %v", boolVal, err)
```
Comment (nkunal) [inline pkg/segment/writer/segwriter.go:369]: we do not want to log every bool that failed, will overwhem the logs

File: pkg/segment/writer/segwriter.go [hunk 2]
```diff
@@ -356,6 +356,23 @@ func (ss *SegStore) doLogEventFilling(ple *ParsedLogEvent, tsKey *string) (bool,
 				ss.checkAddDictEnc(colWip, colWip.cbuf[startIdx:colWip.cbufidx], ss.wipBlock.blockSummary.RecCount, startIdx, false)
 			}
 			ss.updateColValueSizeInAllSeenColumns(cname, colWip.cbufidx-startIdx)
+		case VALTYPE_ENC_BOOL[0]:
+			startIdx := colWip.cbufidx
+			copy(colWip.cbuf[startIdx:], ple.allCvalsTypeLen[i][0:1])
+			colWip.cbufidx += 1
+			copy(colWip.cbuf[colWip.cbufidx:], ple.allCvals[i][0:1])
+			colWip.cbufidx += 1
```
Comment (AndrewHess) [inline pkg/segment/writer/segwriter.go:364]: nit: just do one `copy()` to copy both bytes

File: pkg/segment/writer/segwriter.go [hunk 3]
```diff
@@ -356,6 +356,23 @@ func (ss *SegStore) doLogEventFilling(ple *ParsedLogEvent, tsKey *string) (bool,
...
+			boolVal := utils.BytesToBoolLittleEndian(ple.allCvals[i][0:1])
+			var asciiBytesBuf bytes.Buffer
+			_, err := fmt.Fprintf(&asciiBytesBuf, "%v", boolVal)
+			if err != nil {
+				return false, utils.TeeErrorf("doLogEventFilling: cannot write boolVal %v: %v", boolVal, err)
+			}
+			addSegStatsBool(segstats, cname, asciiBytesBuf.Bytes())
```
Comment (AndrewHess) [inline pkg/segment/writer/segwriter.go]: Why do you have to do so much work here to get `asciiBytesBuf.Bytes()`? Isn't it always going to be 1?

File: pkg/segment/writer/segwriter.go (hunk 2 above)
Comment (sunitakawane) [reply pkg/segment/writer/segwriter.go:364]: I am copying from two different sources.

File: pkg/segment/writer/segwriter.go (hunk 3 above)
Comment (sunitakawane) [reply pkg/segment/writer/segwriter.go]: bool value can be true or false , we need to get the value not just the length

File: pkg/segment/writer/segwriter.go (hunk 2 above)
Comment (AndrewHess) [reply pkg/segment/writer/segwriter.go:364]: Oh, my bad

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**